portfolio/src/main/webapp/calendar.js

```diff
@@ -151,15 +152,180 @@ function collectAllEvents() {
+ * The default end time is set to the end of working hour.
+ */
+function setNewEventStartAndEndTimes() {
+  const workHourStartString = $('#working-hour-start').val();
+  const workHourEndString = $('#working-hour-end').val();
+
+  const userPickedDate = getUserPickedDate();
+  var defaultEventStartHour; 
+
+  if (isToday(userPickedDate)) {
+    // If user picked today to schedule for, 
+    // the default event starts on the next hour if it is before
+    // working hour ends. 
+    const now = new Date();
+    var nextHour = now.getHours() + 1;
+    defaultEventStartHour = getClosestNextHour(nextHour, workHourStartString, workHourEndString);
+  } else {
+    defaultEventStartHour = workHourStartString; 
+  }
+   
+  $('#new-event-start-time').val(defaultEventStartHour);
+  $('#new-event-end-time').val(workHourEndString);
+}
+
+/** Checks if a date is today. */
+function isToday(date) {
+  const today = new Date();
+  return (date.getFullYear() === today.getFullYear()
+      && date.getMonth() === today.getMonth()
+      && date.getDate() === today.getDate());
+}
+
+/** 
+ * Gets the closest next hour compared to the current time.
+ * This hour is displayed as the placeholder for creating new event's 
+ * start time. 
+ * If the closest hour is before working hour start time, sets it to
+ * working hour start time. If the closest hour is after working hour end
+ * time, sets it to working hour end time. 
+ * This function is called if the user picks today to schedule for. 
+ * @return closest next hour in HH:00 format. 
+ */
+function getClosestNextHour(nextHour, workHourStartString, workHourEndString) {
+  if (nextHour == 24) {
+    nextHour = 0; 
+  }
+
+  const workHourStart = parseInt(workHourStartString.split(':')[0]);
+  const workHourEnd = parseInt(workHourEndString.split(':')[0]);
+
+  var closestHour = nextHour; 
+  if (nextHour <= workHourStart) {
+    closestHour = workHourStartString;
+  } else if (nextHour >= workHourEnd) {
+    closestHour = workHourEndString;
+  } else {
+    if (nextHour < 10) {
+      closestHour = '0' + nextHour;
+    }
+    closestHour += ':00';
+  }
+
+  return closestHour; 
+}
+ 
+/** Checks the validity of the user's working hours input. */
+function checkWorkingHourRange() {
+  const workHourStartParts = $('#working-hour-start').val().split(':'); 
+  const workHourStartHour =
+      parseInt(workHourStartParts[0]);
+  const workHourStartMinute =
+      parseInt(workHourStartParts[1]);
+
+  const workHourEndParts = $('#working-hour-end').val().split(':'); 
+  const workHourEndHour = parseInt(workHourEndParts[0]);
+  const workHourEndMinute =
+      parseInt(workHourEndParts[1]);
+  
+  const $workHourWarning = $('#working-hour-warning'); 
+  if (!isWorkingHourValid(
+        workHourStartHour, workHourEndHour, workHourStartMinute, workHourEndMinute)) {
+    $workHourWarning.removeClass('d-none').text('Working hours are not valid.');
+  } else {
+    $workHourWarning.addClass('d-none');
```

Comment:
Consider emptying the text as well.

$workHourWarning.empty().addClass(...)

Reply:
Updated.